### ISSUE TITLE
Bookmarking

### ISSUE THREAD

**Issue by MasterOfTheTiger** [heb12/heb12#7]
One thing that we need to add is the ability to bookmark chapters. It is a great feature, and I propose that it would work like this:
- You can bookmark an unlimited about of chapters
- There is a bookmark icon on the top right corner of the result element
- In the header there should be a dropdown (custom not `<select>`) that lets you select bookmarked chapters

**Comment by edxtech** [comment]
Cookie or cloud?

**Comment by MasterOfTheTiger** [comment]
@edxtech localStorage is what I have been using. Eventually (when we get the backend set up) we can do cloud.

**Comment by amazinigmech2418** [comment]
This is already active with local storage.

**Comment by amazinigmech2418** [comment]
Never mind. I thought this was to save single chapters, not multiple.

**Comment by MasterOfTheTiger** [comment]
I think we should have it available by the initial release.

**Comment by doodthedoodthedood** [comment]
Should I do it on the local storage?

**Comment by MasterOfTheTiger** [comment]
Yes. Make it have a bookmark button in the right corner of the page, and make it save it into an array with a list of chapters that are bookmarked. Then make a place to view them and click them (setChapter() will be useful here).

**Comment by MasterOfTheTiger** [comment]
@doodthedoodthedood How is it going?

**Comment by doodthedoodthedood** [comment]
I have not had a chance to start it yet, but this friday and satruday I will have some time. Also probably tomorrow.

**Comment by MasterOfTheTiger** [comment]
Awesome! I was just making sure I had someone on it. I will officially assign the issue to you.

**Comment by doodthedoodthedood** [comment]
Okay! Great!

**Comment by doodthedoodthedood** [comment]
I won't. I will try my best! Hopefully it should be done by next wednesday. As for to bookmark, should the bookmark icon to bookmark it look like the one at the top right corner of the comment box, outlined if not bookmarked, filled in if it is?

**Comment by MasterOfTheTiger** [comment]
Yes. That is exactly what I was thinking. You can use font awesome if you would like. They probably have something.

**Comment by MasterOfTheTiger** [comment]
You can make another branch in this repo and work on it if you want.

**Comment by doodthedoodthedood** [comment]
Okay, and also should I make it so it is a different color for each theme?

Also I can not work on it now, but tomorrow at around 2 I should have an hour or two to work on it.

**Comment by doodthedoodthedood** [comment]
This is offtopic but how do I save my work in a new branch without making a pull request?

**Comment by MasterOfTheTiger** [comment]
So use the Git client and do `git branch bookmarking` (that creates the new branch on your computer) and then do `git checkout bookmarking` (that changes to that branch). I recommend a text editor like Atom that has Git support built in. Also, it would be best if you reclone the repo from the main repo instead of your fork if you want to use a branch here.

**Comment by MasterOfTheTiger** [comment]
A different color for each theme would be awesome! I also think we should rethink how themes work, they don’t seem to change too much.

**Comment by MasterOfTheTiger** [comment]
No rush. Do it when you can.

**Comment by doodthedoodthedood** [comment]
Great! Also could I still use notepad++. I am just familiar with the interface for it.

**Comment by MasterOfTheTiger** [comment]
I don’t care. I think it would be useful for you to use something with built in Git support. Does notepad ++ have Git support built in or in an add on?

**Comment by doodthedoodthedood** [comment]
No, it does not. But I can use github desktop, which I think does.

**Comment by MasterOfTheTiger** [comment]
GitHub desktop will work.

**Comment by doodthedoodthedood** [comment]
Great!

**Comment by MasterOfTheTiger** [comment]
I don’t use it because I use Linux, but Atom is made by GitHub and has Git built in so that’s why I use it.

**Comment by doodthedoodthedood** [comment]
Yeah, I get it. I just don't really want to download any more text editors XD

**Comment by MasterOfTheTiger** [comment]
Lol. Okay I don’t care. As long as you are using something that works.

**Comment by doodthedoodthedood** [comment]
Lol. I have to go now, but I will talk to you tomorrow about the progress!

**Comment by doodthedoodthedood** [comment]
I did that responding to the email I got sent

**Comment by MasterOfTheTiger** [comment]
Okay @doodthedoodthedood, when do you think you can start working on it?

**Comment by doodthedoodthedood** [comment]
I am working on it, but it will be a little harder then I thought. I will submit a pull request when I think it is good, but no, I am not working on it using github.

**Comment by MasterOfTheTiger** [comment]
@doodthedoodthedood That's fine, just do it when you can.

**Comment by doodthedoodthedood** [comment]
Okay, thanks

**Comment by doodthedoodthedood** [comment]
#

**Comment by doodthedoodthedood** [comment]
Oops wrong button XD

**Comment by MasterOfTheTiger** [comment]
Wow. I don't think this is finished yet.

**Comment by doodthedoodthedood** [comment]
Neither do I XD

**Comment by MasterOfTheTiger** [comment]
@doodthedoodthedood Hello. So how is it going?

**Comment by MasterOfTheTiger** [comment]
@doodthedoodthedood Okay, so I guess I might have to work on this later (in a few weeks). Can you start development?

**Comment by amazinigmech2418** [comment]
We need to use local storage for the (currently down) online alpha.

**Comment by doodthedoodthedood** [comment]
@MasterOfTheTiger sorry about inactivity. I have a lot of camps this summer and am very busy. I will get some work done now.

**Comment by doodthedoodthedood** [comment]
I have to start fresh because I wiped my computer clean and forgot to back up the work

**Comment by doodthedoodthedood** [comment]
nvm gtg
